libs/designer-ui/src/lib/utils/utils.ts

```diff
@@ -113,6 +113,19 @@ export function getDurationString(milliseconds: number, abbreviated = true): str
   }
 }
 
+/**
+ * Returns a string with a duration, possibly abbreviated, e.g., 15s or 15 second(s)
+ * @arg {string} startTime - The start time of the duration
+ * @arg {string} endTime - The end time of the duration
+ * @arg {boolean} [abbreviated=true] - True if the string should be abbreviated, e.g., "s" instead of "second(s)".
+ */
+export function getDurationStringFromTimes(startTime: string, endTime: string, abbreviated = true): string {
+  const start = new Date(startTime);
```

Comment:
Does this handle undefined? For running status it won't have endtime

Reply:
This function does not, in the component if start or end are undefined we don't call this to get the duration string